app/lib/system_admin_constraint.rb

```diff
@@ -8,7 +8,17 @@ def matches?(request)
 private
 
   def system_admin?(request)
-    signin_user = DfESignInUser.load_from_session(request.session)
+    signin_user =
+      if otp_feature_enabled?
+        OtpSignInUser.load_from_session(request.session)
+      else
+        DfESignInUser.load_from_session(request.session)
+      end
```

Comment:
Potential edge case: User authenticates with DSI and then we flip the feature flag to OTP. Should that user still be considered authenticated?